org.eclipse.scanning.connector.epicsv3/src/org/eclipse/scanning/connector/epicsv3/EpicsV3DynamicDatasetConnector.java

```diff
@@ -299,6 +274,56 @@ public boolean refreshShape() {
 		return false;
 	}
 	
+	private int getDataSize() {
+		String currentWidthPV;
+		String currentHeightPV;
+		String rgbChannelsPV;
+
+		if (colourMode.equals(monoColourMode)) {
+
+			width = dim0;		currentWidthPV = dim0PV;
+			height = dim1;		currentHeightPV = dim1PV;
+			rgbChannels = 1;	rgbChannelsPV = dim2PV;
+
+		} else {
+			if (colourMode.equals("RGB2")) {
+
+				width = dim0;		currentWidthPV = dim0PV;
+				height = dim2;		currentHeightPV = dim2PV;
+				rgbChannels = dim1;	rgbChannelsPV = dim1PV;
+
+			} else if (colourMode.equals("RGB3")) {
+
+				width = dim0;		currentWidthPV = dim0PV;
+				height = dim1;		currentHeightPV = dim1PV;
+				rgbChannels = dim2;	rgbChannelsPV = dim2PV;
+
+			} else {
+
+				width = dim1;		currentWidthPV = dim1PV;
+				height = dim2;		currentHeightPV = dim2PV;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)